GitCommands/Repository/RepositoryShortNameProvider.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.IO;
+using System.Linq;
+using GitCommands.Git;
+
+namespace GitCommands.Repository
+{
+    public interface IRepositoryShortNameProvider
+    {
+        /// <summary>
+        /// Returns a short name for repository.
+        /// If the repository contains a description it is returned,
+        /// otherwise the last part of path is returned.
+        /// </summary>
+        /// <param name="repositoryDir">Path to repository.</param>
+        /// <returns>Short name for repository</returns>
+        string Get(string repositoryDir);
+    }
+
+    public sealed class RepositoryShortNameProvider : IRepositoryShortNameProvider
+    {
+        private const string RepositoryDescriptionFileName = "description";
+        private const string DefaultDescription = "Unnamed repository; edit this file 'description' to name the repository.";
+        private readonly IGitDirectoryResolver _gitDirectoryResolver;
+
+
+        public RepositoryShortNameProvider(IGitDirectoryResolver gitDirectoryResolver)
+        {
+            _gitDirectoryResolver = gitDirectoryResolver;
+        }
+
+
+        /// <summary>
+        /// Returns a short name for repository.
+        /// If the repository contains a description it is returned,
+        /// otherwise the last part of path is returned.
+        /// </summary>
+        /// <param name="repositoryDir">Path to repository.</param>
+        /// <returns>Short name for repository</returns>
+        public string Get(string repositoryDir)
```

Comment:
This should be the repo description, not repo shortname.
.git/description does not seem to be cloned.

I consider the repo shortname as the name part of the repo. Used like that in #4208

Reply:
Good point. I took the original method's name without really looking at the underlying functionality.